test/query_test.rb

```diff
@@ -32,9 +39,21 @@ def test_find
     assert_equal [], User.find(name: "Noah").to_a
   end
 
+  def test_find!
+    user = User.create(name: "Bryan")
+    assert_equal user, User.find!(user.id)
+    assert_raises(Minidoc::DocumentNotFoundError) { User.find!(BSON::ObjectId.new) }
```

Comment:
Ah, I totally missed that point. I'll give that some thought, but I'm leaning towards maybe only supporting `find_one!` since that subtle difference between `find!` with an `id` vs a `selector` feels pretty surprising.